File: visual_basic/integracion_con_base_de_datos/eliminar_contacto/eliminar_contacto.vb
Imports System
Imports MySql.Data.MySqlClient
Module eliminar_contacto
    Sub Main()
        Dim opcion As String = ""
        Do
            Console.Clear()
            Console.WriteLine("📒 MENÚ DE CONTACTOS")
            Console.WriteLine("1. ➕ Agregar contacto")
            Console.WriteLine("2. 📋 Mostrar contactos")
            Console.WriteLine("3. ❌ Eliminar contacto por ID")
            Console.WriteLine("4. 🚪 Salir")
            Console.Write("Selecciona una opción: ")
            opcion = Console.ReadLine()
            Select Case opcion
                Case "1"
                    AgregarContacto()
                Case "2"
                    MostrarContactos()
                Case "3"
                    EliminarContacto()
                Case "4"
                    Console.WriteLine("👋 Saliendo...")
                Case Else
                    Console.WriteLine("⚠ Opción no válida")
            End Select
            If opcion <> "4" Then
                Console.WriteLine(vbCrLf & "Presiona una tecla para continuar...")
                Console.ReadKey()
            End If
        Loop While opcion <> "4"
    End Sub
    Sub AgregarContacto()
        Console.Clear()
        Console.WriteLine("➕ AGREGAR CONTACTO")
        Console.Write("Nombre: ")
        Dim nombre As String = Console.ReadLine()
        Console.Write("Teléfono: ")
        Dim telefono As String = Console.ReadLine()
        Console.Write("Correo: ")
        Dim correo As String = Console.ReadLine()
        Try
            Using conexion As MySqlConnection = ObtenerConexion()
                conexion.Open()
                Dim query As String = "INSERT INTO contactos (nombre, telefono, correo) VALUES (@nombre, @telefono, @correo)"
                Using comando As New MySqlCommand(query, conexion)
                    comando.Parameters.AddWithValue("@nombre", nombre)
                    comando.Parameters.AddWithValue("@telefono", telefono)
                    comando.Parameters.AddWithValue("@correo", correo)
                    comando.ExecuteNonQuery()
                End Using
            End Using
            Console.WriteLine("✅ Contacto guardado correctamente")
        Catch ex As Exception
            Console.WriteLine("❌ Error: " & ex.Message)
        End Try
    End Sub
    Sub MostrarContactos()
        Console.Clear()
        Console.WriteLine("📋 LISTA DE CONTACTOS")
        Console.WriteLine("------------------------")
        Try
            Using conexion As MySqlConnection = ObtenerConexion()
                conexion.Open()
                Dim query As String = "SELECT * FROM contactos"
                Using comando As New MySqlCommand(query, conexion)
                    Using lector As MySqlDataReader = comando.ExecuteReader()
                        If lector.HasRows Then
                            While lector.Read()
                                Console.WriteLine("ID: " & lector("id"))
                                Console.WriteLine("Nombre: " & lector("nombre"))
                                Console.WriteLine("Teléfono: " & lector("telefono"))
                                Console.WriteLine("Correo: " & lector("correo"))
                                Console.WriteLine("------------------------")
                            End While
                        Else
                            Console.WriteLine("⚠ No hay contactos")
                        End If
                    End Using
                End Using
            End Using
        Catch ex As Exception
            Console.WriteLine("❌ Error: " & ex.Message)
        End Try
    End Sub
    Sub EliminarContacto()
        Console.Clear()
        Console.WriteLine("❌ ELIMINAR CONTACTO")
        Console.Write("Ingresa el ID del contacto: ")
        Dim id As Integer = Convert.ToInt32(Console.ReadLine())
        Try
            Using conexion As MySqlConnection = ObtenerConexion()
                conexion.Open()
                Dim query As String = "DELETE FROM contactos WHERE id = @id"
                Using comando As New MySqlCommand(query, conexion)
                    comando.Parameters.AddWithValue("@id", id)
                    Dim filasAfectadas As Integer = comando.ExecuteNonQuery()
                    If filasAfectadas > 0 Then
                        Console.WriteLine("✅ Contacto eliminado correctamente")
                    Else
                        Console.WriteLine("⚠ No se encontró el contacto")
                    End If
                End Using
            End Using
        Catch ex As Exception
            Console.WriteLine("❌ Error: " & ex.Message)
        End Try
    End Sub
End Module

File: python/eliminar_contacto/eliminar_contacto.py
from conexion import conectar
def guardar_contacto(nombre, telefono, email):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "INSERT INTO contactos (nombre, telefono, email) VALUES (%s, %s, %s)"
        valores = (nombre, telefono, email)
        cursor.execute(sql, valores)
        conexion.commit()
        cursor.close()
        conexion.close()
        print("✅ Contacto guardado")
def mostrar_contactos():
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        cursor.execute("SELECT * FROM contactos")
        resultados = cursor.fetchall()
        if resultados:
            print("\n📋 Contactos:\n")
            for contacto in resultados:
                print(f"ID: {contacto[0]} | Nombre: {contacto[1]} | Tel: {contacto[2]}")
        else:
            print("⚠️ No hay contactos")
        cursor.close()
        conexion.close()
def eliminar_contacto(id_contacto):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "DELETE FROM contactos WHERE id = %s"
        cursor.execute(sql, (id_contacto,))
        conexion.commit()
        if cursor.rowcount > 0:
            print("🗑️ Contacto eliminado correctamente")
        else:
            print("⚠️ No se encontró el contacto")
        cursor.close()
        conexion.close()
def main():
    print("=== Eliminar Contacto ===")
    while True:
        print("\n1. Agregar contacto")
        print("2. Mostrar contactos")
        print("3. Eliminar contacto")
        print("4. Salir")
        opcion = input("Elige una opción: ")
        if opcion == "1":
            nombre = input("Nombre: ")
            telefono = input("Teléfono: ")
            email = input("Email: ")
            guardar_contacto(nombre, telefono, email)
        elif opcion == "2":
            mostrar_contactos()
        elif opcion == "3":
            mostrar_contactos()
            try:
                id_contacto = int(input("Ingrese el ID a eliminar: "))
                eliminar_contacto(id_contacto)
            except ValueError:
                print("❌ Debes ingresar un número válido")
        elif opcion == "4":
            print("👋 Saliendo")
            break
        else:
            print("❌ Opción inválida")
if __name__ == "__main__":
    main()